java-client/weather-server-example/src/main/java/io/deephaven/client/examples/ui/Ui.java

```diff
@@ -0,0 +1,253 @@
+        if (flightSession != null) {
+            try {
+                support.close();
+                flightSession.close();
+                managedChannel.shutdownNow();
+                managedChannel.awaitTermination(5, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                // No error handling for now.
+                e.printStackTrace();
+                return;
+            }
+        }
+
+        SwingUtilities.invokeLater(() -> statusLabel.setText("Connecting..."));
+        boolean plaintext = plaintextCheckBox.isSelected();
+
+        // Create the new connection
+        final ManagedChannelBuilder<?> channelBuilder = ManagedChannelBuilder.forTarget(host);
+        if (plaintext || "localhost:10000".equals(host)) {
+            channelBuilder.usePlaintext();
+        } else {
+            channelBuilder.useTransportSecurity();
+        }
+
+        managedChannel = channelBuilder.build();
+
+        final BufferAllocator bufferAllocator = new RootAllocator();
+        final FlightSessionFactory flightSessionFactory =
+                DaggerDeephavenFlightRoot.create().factoryBuilder()
+                        .managedChannel(managedChannel)
+                        .scheduler(flightScheduler)
+                        .allocator(bufferAllocator)
+                        .build();
+
+        SwingUtilities.invokeLater(() -> statusLabel.setText("Connected!"));
+
+        flightSession = flightSessionFactory.newFlightSession();
+        support = new BarrageSupport(managedChannel, flightSession);
+        statsTable = support.fetchSubscribedTable("s/LastByCityState");
+        configureDisplay();
+    }
+
+    private void configureDisplay() {
+        summaryTable.setModel(new BarrageBackedTableModel(statsTable));
+    }
+
+    /**
+     * When the application shuts down, ensure that we clean up our connections
+     * and shutdown any running threads.
+     */
+    private void onShutdown() {
+        if (statsTable != null) {
+            support.releaseTable(statsTable);
+        }
+
+        if (support != null) {
+            support.close();
+        }
+
+        flightScheduler.shutdown();
+        try {
+            if (!flightScheduler.awaitTermination(10, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Scheduler not shutdown after 10 seconds");
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return;
+        }
+
+        if (managedChannel != null) {
+            managedChannel.shutdown();
+            try {
+                if (!managedChannel.awaitTermination(10, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Channel not shutdown after 10 seconds");
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    {
+// GUI initializer generated by IntelliJ IDEA GUI Designer
```

Comment:
If IJ regenerates this on every execution, you probably want to mark the rest of this file as:
```
// @formatter:off
[ ... ]
// @formatter:on
```